package/src/engines/vu_meter_worker.ts

```diff
@@ -0,0 +1,33 @@
+/*
+  Copyright 2022 Picovoice Inc.
+
+  You may not use this file except in compliance with the license. A copy of the license is located in the "LICENSE"
+  file accompanying this source.
+
+  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+  specific language governing permissions and limitations under the License.
+*/
+
+import { WvpMessageEvent } from '../types';
+
+const INT_16_MAX = 32767;
+const EPSILON = 1e-9;
+
+const process = (frames: Int16Array): number => {
+  const sum = [...frames].reduce(
+    (accumulator, frame) => accumulator + frame ** 2,
+    0,
+  );
+  const rms = Math.sqrt(sum / frames.length) / INT_16_MAX;
```

Comment:
as we discussed if you'd like you can optimize this part for a better run-time speed by skipping the `sqrt` function and multiply the result by 10, instead of 20; but the math checks out as is